lib/isolator/orm_adapters/active_support_subscriber.rb

```diff
@@ -9,8 +9,9 @@ module ActiveSupportSubscriber
 
     def self.subscribe!(event)
       ::ActiveSupport::Notifications.subscribe(event) do |_name, _start, _finish, _id, query|
-        Isolator.incr_transactions! if START_PATTERN.match?(query[:sql])
-        Isolator.decr_transactions! if FINISH_PATTERN.match?(query[:sql])
+        connection_id = query[:connection_id] || query[:connection]&.object_id || 0
```

Comment:
I'm curious how this works given [this comment](https://github.com/palkan/isolator/pull/29#issuecomment-636392924)

Reply:
`ActiveRecord::Base.connection` retrieves a connection from the pool. If you call it from within a subscriber or in a transaction test, then it's gonna be a different connection (because the current one, holding a transaction, is busy). I think, that's the reason.